src/Controls/tests/UITests/Tests/Issues/Issue7823.cs

```diff
@@ -0,0 +1,32 @@
+﻿using NUnit.Framework;
+using OpenQA.Selenium;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.AppiumTests.Issues
+{
+	public class Issue7823 : _IssuesUITest
+	{
+		public Issue7823(TestDevice device)
+			: base(device)
+		{ }
+
+		public override string Issue => "In a ToolbarItems, if an item has no icon but just text, MAUI uses the icon from the previous page in the Navigation";
+
+		[Test]
+		public async Task UpdateToolbarItemAfterNavigate()
+		{
+			this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.iOS, TestDevice.Mac, TestDevice.Windows });
+
+			// 1. Navigate from Page with a TollbarItem using an Icon. 
+			App.WaitForElement("WaitForStubControl");
+			App.Click("WaitForStubControl");
+
+			await Task.Delay(1000); // Wait navigation animation to complete
```

Comment:
@jsuarezruiz instead of waiting 1000 can we just call "WaitForElement" on this label? https://github.com/dotnet/maui/pull/16796/files#diff-519313869e097359185cfad4f74168f6e2b14c226a155493eec26df3227e37d1R11-R14